src/main/java/org/opensearch/knn/index/util/ScheduledExecutor.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.util;
+
+import lombok.Getter;
+
+import java.io.Closeable;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Executes a task periodically
+ */
+public class ScheduledExecutor implements Closeable {
+    final ScheduledExecutorService executor;
+    @Getter
+    private final Runnable task;
+
+    /**
+     * @param task task to be completed
+     * @param scheduleMillis time in milliseconds to wait before executing the task again
+     */
+    public ScheduledExecutor(Runnable task, long scheduleMillis) {
+        this.task = task;
+        this.executor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
This seems to be creating a new thread for each instance, I wanted to understand the thought process behind this.

I understand that this is used for cache so the threads won't grow exponentially, but since its in util package and the name is generic - any misuse of this can create unbounded number of threads with each instance of this. Can we be a bit more defensive here?

The suggestion here is to refactor and make sure there are fixed number of threads for cache cleanup across caches

Reply:
Thanks, I updated ScheduledExecutor to accept an external executor and modified each cache to create and provide their own single threaded executor.